### ISSUE TITLE
Generate Source from az cli call

### ISSUE THREAD

**Issue by git001** [Azure/azure-cli#13192]
**Is your feature request related to a problem? Please describe.**  
The problem which I try to resolve is the dynamic assign of Egress-IPs in OpenShift on Azure.   
OCP 3.11: https://docs.openshift.com/container-platform/3.11/admin_guide/managing_networking.html#admin-guide-automatic-egress-ip   
OCP 4: https://docs.openshift.com/container-platform/4.3/networking/openshift_sdn/assigning-egress-ips.html#nw-egress-ips-automatic_egress-ips  
My current solution is manually assign the IP to a NIC which is not really HA friendly
```
az network nic ip-config create \
  --name ocpcp003-nic-egressip \
  --nic-name ocpcp003-nic \
  --resource-group ocp-rg \
  --private-ip-address <THE-EGRESSIP> \
  --private-ip-address-version IPv4 \
  --subscription <your-subscription>
```

**Describe the solution you'd like**
I would like to contribute to https://github.com/redhat-cop/egressip-ipam-operator which is written in golang. My Feature request is about two paramters in the cli which is able to create a source code from the **az cli** similar to [curl --libcurl](https://curl.haxx.se/docs/manpage.html#--libcurl) which creates a source code file from curls cli call.

My suggestion is the following.
```
az network nic ip-config create \
  --name ocpcp003-nic-egressip \
  --nic-name ocpcp003-nic \
  --resource-group ocp-rg \
  --private-ip-address <THE-EGRESSIP> \
  --private-ip-address-version IPv4 \
  --subscription <your-subscription> \
  --generate-code my-source-file.go \ # new parameter 1
  --programming-language golang # new parameter 1
```

**Describe alternatives you've considered**
Not alternatives considered

**Additional context**

I have tried to use the https://github.com/Azure-Samples/azure-sdk-for-go-samples/blob/master/network/ipaddress.go with https://github.com/Azure/azure-sdk-for-go and https://godoc.org/github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-11-01/network#NewInterfaceIPConfigurationsClient but I was not able to use the interface.

**Comment by yonzhan** [comment]
add to S170

**Comment by MyronFanQiu** [comment]
add @jiasli  and @Juliehzl  for awareness.

**Comment by MyronFanQiu** [comment]
@git001 Did you try terraform or ansible?

**Comment by git001** [comment]
@MyronFanQiu nope because I need golang code to implement the ip-config create command

**Comment by jiasli** [comment]
Azure CLI is written in Python. Besides it is impracticable to include all languages like Go, Ruby, C, etc. Also, [curl --libcurl](https://curl.haxx.se/docs/manpage.html#--libcurl) only supports C. 😉

You may be interested in [Azure SDK for Go](https://github.com/Azure/azure-sdk-for-go).

Another solution is to dump HTTP requests with `--debug` and make an HTTP call with any tools you like, including `az rest`:

```
> az group list --debug

msrest.http_logger : Request URL: 'https://management.azure.com/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourcegroups?api-version=2019-07-01'
msrest.http_logger : Request method: 'GET'
msrest.http_logger : Request headers:
msrest.http_logger :     'Accept': 'application/json'
msrest.http_logger :     'accept-language': 'en-US'
msrest.http_logger :     'User-Agent': 'python/3.8.1 (Windows-10-10.0.18362-SP0) msrest/0.6.10 msrest_azure/0.6.3 azure-mgmt-resource/9.0.0 Azure-SDK-For-Python AZURECLI/2.5.0 (PIP)'
```

 Let's me know if this suits your needs.

**Comment by git001** [comment]
@jiasli "at libcurl" I know ;-)  

I'm on the way to learn the go sdk but it's quite difficult because I'm new to go.  
Okay so you mean we should close this issue?

**Comment by jiasli** [comment]
Hi @git001, I agree that SDK may be difficult to start with. 

If the Azure management call is a one-time operation, I will go with the [http](https://golang.org/pkg/net/http/) REST API approach, which is more general and not language-specific. As for the authorization and authentication part, we recommend [using a Service Principal](https://docs.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli?view=azure-cli-latest), which can be created with `az ad sp create-for-rbac`, then follow [OAuth 2.0 client credentials flow](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-client-creds-grant-flow) to get the access token for the request. 

Another easier solution is to subprocessing `az` and call it from your program with [exec](https://golang.org/pkg/os/exec/). 

So to summarize we have 3 approaches:

1. [Azure SDK for Go](https://github.com/Azure/azure-sdk-for-go)
2. [http](https://golang.org/pkg/net/http/)
3. [exec](https://golang.org/pkg/os/exec/)

Hope it helps.

**Comment by git001** [comment]
thanks for the answers.
